File: process-redirects.py
# Process CSV redirects and generate output files
import csv
import re

csv_path = r'C:\Users\VIKASH\Downloads\eazybe 301 redirection file - Sheet1.csv'
app_output_path = r'C:\Users\VIKASH\websanity\src\routes\redirect-routes.tsx'
nginx_output_path = r'C:\Users\VIKASH\websanity\nginx-redirects.conf'
redirects_output_path = r'C:\Users\VIKASH\websanity\public\_redirects'

# Read CSV file
with open(csv_path, 'r', encoding='utf-8') as f:
    csv_content = f.read()
    lines = csv_content.split('\n')

# Parse redirects
redirects = []
english = []
portuguese = []
spanish = []
turkish = []

for i, line in enumerate(lines):
    line = line.strip()
    if not line or line.startswith('Old URL'):
        continue

    parts = line.split(',')
    if len(parts) < 2:
        continue

    source = parts[0].replace('Old URL (Source)', '').strip()
    dest = parts[1].replace('New URL (Destination)', '').strip()
    status = parts[2].replace('Status', '').strip() if len(parts) > 2 else '301'

    # Clean source - remove trailing comma and slash
    source = re.sub(r',+$', '', source)
    source = re.sub(r'/$', '', source)

    # Skip if empty or status is not 301
    if not source or not dest or status != '301':
        continue

    redirects.append({'source': source, 'dest': dest, 'status': status})

print(f'Total redirects parsed: {len(redirects)}')

# Group by language prefix
for r in redirects:
    s = r['source']
    if s.startswith('/br/'):
        portuguese.append(r)
    elif s.startswith('/es/'):
        spanish.append(r)
    elif s.startswith('/tr/'):
        turkish.append(r)
    else:
        english.append(r)

# Generate App.tsx JSX routes
def generate_redirect_route(src, dest):
    src = re.sub(r'/$', '', src)
    dest = re.sub(r'/$', '', dest)

    has_lang_prefix = dest.startswith('/br/') or dest.startswith('/es/') or dest.startswith('/tr/')

    return f'      <Route path="{src}" element={{{<Navigate to="{dest}" replace />}}'

def generate_routes(arr, lang_prefix=''):
    return '\n'.join(generate_redirect_route(r['source'], r['dest']) for r in arr)

english_routes = generate_routes(english)
portuguese_routes = generate_routes(portuguese)
spanish_routes = generate_routes(spanish)
turkish_routes = generate_routes(turkish)

# Generate nginx rules
def generate_nginx_rule(src, lang_prefix=''):
    src = re.sub(r'/$', '', src)
    path = f'{lang_prefix}{src}' if lang_prefix else src

    return f'    rewrite ^{src}$ {path} permanent;'

english_nginx = '\n'.join(generate_nginx_rule(r['source'], r['dest']) for r in english)
portuguese_nginx = '\n'.join(generate_nginx_rule(r['source'], r['dest'], '/br/') for r in portuguese)
spanish_nginx = '\n'.join(generate_nginx_rule(r['source'], r['dest'], '/es/') for r in spanish)
turkish_nginx = '\n'.join(generate_nginx_rule(r['source'], r['dest'], '/tr/') for r in turkish)

# Generate _redirects format
all_redirects = '\n'.join(f'{r["source"].replace("/", "")} {r["dest"].replace("/", "")} 301' for r in redirects)

# Write outputs
app_routes_output = f'''{{/* 301 Redirects */}}

{{' * 8203; 301 Redirects - English */}}
{english_routes}

{{' * 8203; 301 Redirects - Portuguese (/br) */}}
{portuguese_routes}

{{' * 8203; 301 Redirects - Spanish (/es) */}}
{spanish_routes}

{{' * 8203; 301 Redirects - Turkish (/tr) */}}
{turkish_routes}}
'''

nginx_output = f'''# 301 Redirects from CSV
{english_nginx}

# Portuguese (/br) redirects
{portuguese_nginx}

# Spanish (/es) redirects
{spanish_nginx}

# Turkish (/tr) redirects
{turkish_nginx}
'''

# Write to files
with open(app_output_path, 'w', encoding='utf-8') as f:
    f.write(app_routes_output)

with open(nginx_output_path, 'w', encoding='utf-8') as f:
    f.write(nginx_output)

with open(redirects_output_path, 'w', encoding='utf-8') as f:
    f.write(all_redirects)

print(f'App routes written to: {app_output_path}')
print(f'Nginx rules written to: {nginx_output_path}')
print(f'_redirects file written to: {redirects_output_path}')

File: process-redirects.js
// Parse CSV redirects and generate output for App.tsx
const fs = require('fs');
const path = require('path');

const csvPath = 'C:\\Users\\VIKASH\\Downloads\\eazybe 301 redirection file - Sheet1.csv';
const outputPath = 'C:\\Users\\VIKASH\\websanity\\redirects-output.txt';

// Read CSV file
const csvContent = fs.readFileSync(csvPath, 'utf8');
const lines = csvContent.split('\n');

// Skip header and parse
const redirects = [];
for (let i = 1; i < lines.length; i++) {
  const line = lines[i].trim();
  if (!line || line.startsWith('Old URL')) continue; // Skip header and empty lines

  // Parse: source,destination,status
  const parts = line.split(',');
  if (parts.length >= 2) {
    let source = parts[0].replace('Old URL (Source)', '').trim();
    let dest = parts[1].replace('New URL (Destination)', '').trim();
    const status = parts.length > 2 ? parts[2].replace('Status', '').trim() : '301';

    // Clean source - remove trailing comma and slash if exists
    if (source.endsWith(',') || source.endsWith('/')) {
      source = source.slice(0, -1);
    }

    // Skip if empty or status is not 301
    if (source && dest && status === '301') {
      redirects.push({ source, dest, status });
    }
  }
}

console.log(`Total redirects parsed: ${redirects.length}`);

// Group by language prefix
const english = [];
const portuguese = [];
const spanish = [];
const turkish = [];
const noLang = [];

for (const r of redirects) {
  const s = r.source;
  if (s.startsWith('/br/')) {
    portuguese.push(r);
  } else if (s.startsWith('/es/')) {
    spanish.push(r);
  } else if (s.startsWith('/tr/')) {
    turkish.push(r);
  } else {
    noLang.push(r);
  }
}

// Generate App.tsx JSX routes
function generateRoutes(arr, langPrefix = '') {
  return arr.map(r => {
    // Clean source and destination - remove trailing slash and comma
    const src = r.source.replace(new RegExp('/+$', 'g'), '').replace(new RegExp('/$', 'g'), '').replace(new RegExp(',+$', 'g'), '').replace(new RegExp('/+$', 'g'), '');
    const dst = r.dest.replace(/,\/g, '').replace(/\/$/, '');

    // Check if destination already has language prefix
    const hasLangPrefix = dst.startsWith('/br/') || dst.startsWith('/es/') || dst.startsWith('/tr/');

    // Use exact path as provided in CSV
    const path = hasLangPrefix ? dst : `/${langPrefix}${dst.replace(/^\//, '')}`;

    return `      <Route path="${src}" element={<Navigate to="${path}" replace />} />`;
  }).join('\n      ');
}

const englishRoutes = generateRoutes(english);
const portugueseRoutes = generateRoutes(portuguese, '/br/');
const spanishRoutes = generateRoutes(spanish, '/es/');
const turkishRoutes = generateRoutes(turkish, '/tr/');

// Generate nginx rules
function generateNginxRules(arr, langPrefix = '') {
  return arr.map(r => {
    let src = r.source.replace(new RegExp('/+$', 'g'), '').replace(new RegExp('/$', 'g'), '');
    const dst = r.dest.replace(/,\/g, '').replace(/\/$/, '');

    // nginx needs exact path match with $
    const path = langPrefix ? `${langPrefix}${src.replace(/^\//, '')}` : src;

    return `    rewrite ^${path}$ ${langPrefix}${dst.replace(/^\//, '')} permanent;`;
  }).join('\n');
}

const englishNginx = generateNginxRules(english);
const portugueseNginx = generateNginxRules(portuguese, '/br/');
const spanishNginx = generateNginxRules(spanish, '/es/');
const turkishNginx = generateNginxRules(turkish, '/tr/');

// Generate _redirects format
function generateRedirectsFile(arr) {
  return arr.map(r => {
    const src = r.source.replace(/,\/g, '').replace(/\/$/, '');
    const dst = r.dest.replace(/,\/g, '').replace(/\/$/, '');
    return `${src} ${dst} 301`;
  }).join('\n');
}

const allRedirects = generateRedirectsFile(redirects);

// Write outputs
const output = `
// ==================== ENGLISH ROUTES ====================
${englishRoutes}

// ==================== PORTUGUESE (/br/) ROUTES ====================
${portugueseRoutes}

// ==================== SPANISH (/es/) ROUTES ====================
${spanishRoutes}

// ==================== TURKISH (/tr/) ROUTES ====================
${turkishRoutes}

// ==================== NGINX RULES ====================
${englishNginx}
${portugueseNginx}
${spanishNginx}
${turkishNginx}

// ==================== _redirects FILE ====================
${allRedirects}
`;

fs.writeFileSync(outputPath, output, 'utf8');
console.log('Output written to:', outputPath);
